### PR TITLE
CD-70932 Add TextSanitizer

### PR DESCRIPTION
**Pull request by johnny-lai**
* Remove CoupaSanitize
* Moved RailsSanitize to XssTerminate::RailsSanitize
* Adjusted xss_terminate options

- [x] @vikascoupa 
- [x] @saghaulor 
- [x] @edk 
- [x] @cmayor 

Additional Behavior Changes:
* It uses the read/write accessors now instead of the `[]`. This means it should start working with attr_encrypted.

### CODE REVIEW

**Comment by johnny-lai** [comment]
Be sure to review the behavior of `TextSanitizer` which is summarised in `text_sanitizer_test.rb`

**Comment by edk** [comment]
cc @cmayor

**Comment by edk** [comment]
👍

**Comment by cmayor** [comment]
👍

**Comment by vikascoupa** [comment]
👍

File: lib/xss_terminate/active_record.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
+module XssTerminate
+  module ActiveRecord
+    def self.included(base)
+      base.extend(ClassMethods)
+      base.send(:include, InstanceMethods)
```
Comment (saghaulor) [inline lib/xss_terminate/active_record.rb]:
```ruby
base.include InstanceMethods
```

File: lib/xss_terminate/active_record.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            attr_options = {}
+          end
+          self.xss_terminate_options = self.xss_terminate_options.merge(a => attr_options)
+        end
+      end
+
+      # Retrieves xss_terminate options for the specified attribute
+      def xss_terminate_options_for(attribute)
+        h = self.xss_terminate_options[attribute]
+        h || ::XssTerminate.default_options.options
```
Comment (saghaulor) [inline lib/xss_terminate/active_record.rb]: `default_options.options` is clunky and redundant.

File: xss_terminate.gemspec [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
 Gem::Specification.new do |s|
   s.name = %q{xss_terminate}
-  s.version = "0.5.3"
+  s.version = "0.6.0"
 
   s.required_rubygems_version = Gem::Requirement.new(">= 0") if s.respond_to? :required_rubygems_version=
   s.authors = ["Luke Francl"]
```
Comment (saghaulor) [inline xss_terminate.gemspec]: You should add your name to the list @johnny-lai

File: xss_terminate.gemspec [hunk 4]
```diff
@@ -15,38 +15,26 @@ Gem::Specification.new do |s|
...
-     "test/models/person.rb",
-     "test/schema.rb",
-     "test/setup_test.rb",
-     "test/xss_terminate_test.rb"
-  ]
+
+  file_list = `git ls-files`.split
+  s.files       = file_list
+
   s.homepage = %q{http://github.com/look/xss_terminate}
```
Comment (saghaulor) [inline xss_terminate.gemspec]: we should probably change the homepage.

File: lib/xss_terminate/active_record.rb (hunk 2 above)
Comment (johnny-lai) [inline lib/xss_terminate/active_record.rb]: Yes.. the naming is bad. I couldn't think of a better one though.

The `::XssTerminate.default_options` are options for the xss_terminate gem. It includes:
* html_sanitier: Sanitizer to use for `html`
* text_sanitizer: Sanitizer to use for `text`
* options: There are the default options are that applied to attributes. The default is to enable xss_terminate and to sanitize as text.

**Comment by johnny-lai** [comment]
Travis is already integrated into master

File: lib/xss_terminate/active_record.rb (hunk 2 above)
Comment (johnny-lai) [inline lib/xss_terminate/active_record.rb]: I renamed
* `XssTerminate.default_options` to `XssTerminate.configuration`
* `XssTerminate.default_options.options` to `XssTerminate.configuration.default_options`

**Comment by johnny-lai** [comment]
Fix issue where TextSanitizer was stripping > from non-tags
```
+      'a>b'      => 'a>b',
+      'a/b>'     => 'a/b>',
+      'a</b>/c>' => 'a/c>',
```

**Comment by saghaulor** [comment]
:shipit:

**Pull request merged**